### PR TITLE
feat(cfw): add data source to get the list of CFW domain name groups

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
add data source to get the list of CFW domain name groups
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add data source to get the list of CFW domain name groups
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/cfw" TESTARGS="-run TestAccDataSourceCfwDomainNameGoups_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cfw -v -run TestAccDataSourceCfwDomainNameGroups_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceCfwDomainNameGroups_basic
=== PAUSE TestAccDataSourceCfwDomainNameGroups_basic
=== CONT  TestAccDataSourceCfwDomainNameGroups_basic
--- PASS: TestAccDataSourceCfwDomainNameGroups_basic (323.12s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cfw       323.169s
```

### CODE REVIEW

File: docs/data-sources/cfw_domain_name_groups.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+* `region` - (Optional, String) Specifies the region in which to query the resource.
+  If omitted, the provider-level region will be used.
+
+* `fw_instance_id` - (Required, String) Specifies the firewall instance ID.
+
+* `object_id` - (Required, String) Specifies the protected object ID.
+
+* `key_word` - (Optional, String) Specifies the key word.
+
+* `domain_set_type` - (Optional, String) Specifies the domain set type.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/cfw_domain_name_groups.md]:
```suggestion
* `type` - (Optional, String) Specifies the domain name group type.
```

File: docs/data-sources/cfw_domain_name_groups.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  + **1**: configuration succeeded.
+  + **2**: configuration in progress.
+  + **3**: normal.
+  + **4**: configuration exception - domain group usage.
+
+* `ref_count` - (Optional, Int) Specifies the domain set reference count.
+
+* `enterprise_project_id` - (Optional, String) Specifies the enterprise project ID.
+
+* `set_id` - (Optional, String) Specifies the domain set ID.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/cfw_domain_name_groups.md]:
```suggestion
* `group_id` - (Optional, String) Specifies the domain name group ID.
```

File: docs/data-sources/cfw_domain_name_groups.md (hunk 1 above)
Comment (profoundwu) [reply docs/data-sources/cfw_domain_name_groups.md]: fixed

File: docs/data-sources/cfw_domain_name_groups.md (hunk 2 above)
Comment (profoundwu) [reply docs/data-sources/cfw_domain_name_groups.md]: fixed.

**Pull request merged**